extensions/amp-subscriptions/0.1/entitlement.js

```diff
@@ -45,29 +43,24 @@ export class Entitlement {
    * @param {string} [input.source]
    * @param {string} [input.raw]
    * @param {string} [input.service]
-   * @param {!Array<string>} [input.products]
-   * @param {?string} [input.subscriptionToken]
-   * @param {boolean} [input.loggedIn]
-   * @param {?MeteringData} [input.metering]
+   * @param {boolean} [input.granted]
+   * @param {?GrantReasons} [input.grantReason]
+   * @param {?JsonObject} [input.data]
    */
-  constructor({source, raw = '', service, products = [],
-    subscriptionToken = '', loggedIn = false, metering = null}) {
+  constructor({source, raw = '', service, granted = false,
+    grantReason = '', data}) {
```

Comment:
`data` here throws

`The type information available for this expression is too loose to ensure conformance.`